core/src/main/java/org/infinispan/notifications/impl/AbstractListenerImpl.java

```diff
@@ -179,7 +181,15 @@ protected boolean validateAndAddListenerInvocations(Object listener, AbstractInv
       Listener l = testListenerClassValidity(listener.getClass());
       boolean foundMethods = false;
       builder.setTarget(listener);
-      builder.setSubject(Security.getSubject());
+
+      Subject subject;
+      if (System.getSecurityManager() == null) {
+         subject = Security.getSubject();
```

Comment:
That does not sound right. The org.infinispan.security.Security class is public so anyone can access it. Neither org.infinispan.security.Security#isTrustedClass is right, since anyone can have classes in a package org.infinispan.blabla that even jar signing won't solve.